src/iface/fragments.rs

```diff
@@ -0,0 +1,170 @@
+use storage::Assembler;
+use time::Instant;
+use managed::ManagedSlice;
+
+/// Default timeout duration
+const FRAGMENTATION_TIMEOUT_MS: i64 = 500;
+
+#[allow(dead_code)]
+pub struct Packet<'a, T: 'a> {
```

Comment:
Why is it generic over `T`? We'll never want to defragment anything except `u8`. This just results in silly `T: Clone` bounds on functions.